src/pykeen/nn/representation.py

```diff
@@ -1251,45 +1251,47 @@ class BackfillRepresentation(PartitionRepresentation):
     def __index__(
         self,
         max_id: int,
-        assignment: Iterable[int],
-        base: Hint[Representation] = None,
+        base_ids: Iterable[int],
+        base: HintOrType[Representation] = None,
         base_kwargs: OptionalKwargs = None,
-        embedding_kwargs: OptionalKwargs = None,
+        backfill: HintOrType[Representation] = None,
+        backfill_kwargs: OptionalKwargs = None,
         **kwargs,
     ):
         """Initialize the representation.
 
         :param max_id:
             The total number of entities that need to be embedded
-        :param assignment:
-            An iterable of integer entity indexes that need to be
+        :param base_ids:
+            An iterable of integer entity indexes which are provided through the base representations
         :param base:
-            the base representation, or hints thereof.
+            the base representation, or a hint thereof.
         :param base_kwargs:
             keyword-based parameters to instantiate the base representation
+        :param backfill:
+            the backfill representation, or hints thereof.
+        :param backfill_kwargs:
+            keyword-based parameters to instantiate the backfill representation
         :param kwargs:
             additional keyword-based parameters passed to :meth:`Representation.__init__`. May not contain `max_id`,
             or `shape`, which are inferred from the base representations.
         """
         # import here to avoid cyclic import
         from . import representation_resolver
 
-        assignment = set(assignment)
-        base = representation_resolver.make(base, base_kwargs, max_id=len(assignment))
-        embedding = Embedding(max_id=max_id - base.max_id, shape=base.shape, **(embedding_kwargs or {}))
-
-        _assignments = []
-        base_idx, embedding_idx = 0, 0
-        for i in range(max_id):
-            if i in assignment:
-                _assignments.append((0, base_idx))
-                base_idx += 1
-            else:
-                _assignments.append((1, embedding_idx))
-                embedding_idx += 1
+        base_ids = list(base_ids)
+        base = representation_resolver.make(base, base_kwargs, max_id=len(base_ids))
+        # comment: not all representations support passing a shape parameter
```

Comment:
so far, only a few representations suppor passing a `shape` parameter (`Embedding` & `LowRank` inside this file)

Reply:
that's okay, I think it's fine to limit the backfill representation to using one of these more simple ones. In fact, we might even want to explicitly check that one of these were chosen

Reply:
another option, I intend to add later on, is to have `max_id` / `shape` be in the `__init__` parameters any time, and use that for verifying inferred ones.

Reply:
cf. #982